portafly/src/components/pages/product/CreateProductPage.tsx

```diff
@@ -0,0 +1,197 @@
+import React, {
+  useState,
+  FormEventHandler,
+  FocusEventHandler,
+  useEffect
+} from 'react'
+
+import {
+  PageSection,
+  TextContent,
+  Text,
+  Card,
+  CardBody,
+  Form,
+  FormGroup,
+  TextInput,
+  TextArea,
+  ActionGroup,
+  Button
+} from '@patternfly/react-core'
+import { useTranslation } from 'i18n/useTranslation'
+import { useHistory, Redirect } from 'react-router'
+import { createProduct, NewProduct } from 'dal/products'
+import { useAsync } from 'react-async'
+import { useAlertsContext, useDocumentTitle } from 'components/util'
+import { ValidationException, Validator } from 'utils'
+
+type Validations = Record<string, {
+  validation: 'default' | 'success' | 'error',
```

Comment:
maybe we can simplify the API with something like:

```suggestion
  validation?: boolean,
```

thus, if `undefined` means is not yet validated, and the boolean will define if it was success or error.

Reply:
It's give by PF I'm afraid https://www.patternfly.org/v4/documentation/react/components/textinput#props